src/common/config_parser.h

```diff
@@ -25,12 +25,49 @@ enum struct mode { training, translation, scoring, server };
  */
 class ConfigParser {
 public:
+
+  ConfigParser(cli::mode mode);
+
   ConfigParser(int argc, char** argv, cli::mode mode, bool validate = false)
-      : modeServer_(mode == cli::mode::server),
-        mode_(mode == cli::mode::server ? cli::mode::translation : mode) {
+    : ConfigParser(mode) {
     parseOptions(argc, argv, validate);
   }
 
+  template<typename T>
+  ConfigParser&
+  addOption(const std::string& args,
+            const std::string& group,
+            const std::string& help,
+            const T val) {
+    std::string previous_group = cli_.switchGroup(group);
+    cli_.add<T>(args,help,val);
+    cli_.switchGroup(previous_group);
+    return *this;
+  }
+
+  template<typename T>
+  ConfigParser&
+  addOption(const std::string& args,
```

Comment:
Maybe both `addOption` methods can be replaced with a method returning reference to `cli_` and let the user to call any of `.add<T>` variations?

Reply:
That's an invitation for developers to break things. See my comments on cli::switchGroup below.